dot/parachain/backing/candidate_backing.go

```diff
@@ -5,124 +5,723 @@ package backing
+		return
+	}
+
+	candidateHash := parachaintypes.CandidateHash{
+		Value: common.MustBlake2bHash(scale.MustMarshal(attested.Candidate)),
+	}
+
+	// If the candidate is already backed, issue new misbehaviors and return.
+	if rpState.backed[candidateHash] {
+		issueNewMisbehaviors(subSystemToOverseer, rpState.RelayParent, rpState.Table)
+		return
+	}
+
+	// Mark the candidate as backed.
+	rpState.backed[candidateHash] = true
+
+	// Convert the attested candidate to a backed candidate.
+	backedCandidate := attestedToBackedCandidate(*attested, &rpState.TableContext)
+	if backedCandidate == nil {
+		issueNewMisbehaviors(subSystemToOverseer, rpState.RelayParent, rpState.Table)
+		return
+	}
+
+	paraID := backedCandidate.Candidate.Descriptor.ParaID
+
+	if rpState.ProspectiveParachainsMode.IsEnabled {
+
+		// Inform the prospective parachains subsystem that the candidate is now backed.
+		subSystemToOverseer <- parachaintypes.ProspectiveParachainsMessage{
+			Value: parachaintypes.PPMCandidateBacked{
+				ParaID:        parachaintypes.ParaID(paraID),
+				CandidateHash: candidateHash,
+			},
+		}
+
+		// Backed candidate potentially unblocks new advertisements, notify collator protocol.
+		subSystemToOverseer <- parachaintypes.CollatorProtocolMessage{
+			Value: parachaintypes.CPMBacked{
+				ParaID:   parachaintypes.ParaID(paraID),
+				ParaHead: backedCandidate.Candidate.Descriptor.ParaHead,
+			},
+		}
+
+		// Notify statement distribution of backed candidate.
+		subSystemToOverseer <- parachaintypes.StatementDistributionMessage{
+			Value: parachaintypes.SDMBacked(candidateHash),
+		}
+
+	} else {
+		// TODO: figure out what this comment means by 'avoid cycles'.
+		//
+		// The provisioner waits on candidate-backing, which means
+		// that we need to send unbounded messages to avoid cycles.
+		//
+		// Backed candidates are bounded by the number of validators,
+		// parachains, and the block production rate of the relay chain.
+		subSystemToOverseer <- parachaintypes.ProvisionerMessage{
+			Value: parachaintypes.PMProvisionableData{
+				RelayParent: rpState.RelayParent,
+				ProvisionableData: parachaintypes.ProvisionableData{
+					Value: parachaintypes.PDBackedCandidate(backedCandidate.Candidate.ToCandidateReceipt()),
+				},
+			},
+		}
+	}
+
+	issueNewMisbehaviors(subSystemToOverseer, rpState.RelayParent, rpState.Table)
+}
+
+// issueNewMisbehaviors checks for new misbehaviors and sends necessary messages to the Overseer subsystem.
+func issueNewMisbehaviors(subSystemToOverseer chan<- any, relayParent common.Hash, table Table) {
+	// collect the misbehaviors to avoid double mutable self borrow issues
+	misbehaviors := table.drainMisbehaviors()
+
+	for _, m := range misbehaviors {
+		// TODO: figure out what this comment means by 'avoid cycles'.
+		//
+		// The provisioner waits on candidate-backing, which means
+		// that we need to send unbounded messages to avoid cycles.
```

Comment:
AFAIK in Golang unbounded messages are sent through an unbounded channel and the ongoing process blocks until something on the other side receives them, differently from bounded messages where the bounded channel has a limit of messages it can store so if the channel has capacity the sender can "send/store" messages in the channel and continue the ongoing process. 

So by `we need to send unbounded messages to avoid cycles` I think they want to send a message and blocks until the receiver drain the message from the channel